drivers/base/Datagram_Device.zig

```diff
@@ -40,6 +40,17 @@ pub fn disconnect(dd: Datagram_Device) void {
 
 pub const WriteError = BaseError || error{ Unsupported, NotConnected };
 
+/// Writes then reads a single `datagram` to the device.
+pub fn write_then_read(dd: Datagram_Device, src: []const u8, dst: []u8) WriteError!void {
+    return try dd.write_then_readv(&.{src}, &.{dst});
+}
+
+/// Writes then reads a single `datagram` to the device.
+pub fn write_then_readv(dd: Datagram_Device, write_chunks: []const []const u8, read_chunks: []const []u8) WriteError!void {
+    const write_then_readv_fn = dd.vtable.write_then_readv_fn orelse return error.Unsupported;
```

Comment:
On discord there were several suggestions about how to handle implementations that are incomplete (no write_then_read fn in this case).  It looks like this orelse pattern handles that nicely already?